internal/cli/command/controlplane/debug.go

```diff
@@ -0,0 +1,241 @@
+	logResult("add pipeline/resources/banzaicloud/persistentvolumes.yaml", addFile(tarWriter, "pipeline/resources/banzaicloud/persistentvolumes.yaml", fetchContainerCommandOutputAndError(options.cpContext, []string{"kubectl", "get", "persistentvolumes", "-oyaml", "-nbanzaicloud"}, env)))
+	logResult("add pipeline/resources/banzaicloud/persistentvolumeclaims.yaml", addFile(tarWriter, "pipeline/resources/banzaicloud/persistentvolumeclaims.yaml", fetchContainerCommandOutputAndError(options.cpContext, []string{"kubectl", "get", "persistentvolumeclaims", "-oyaml", "-nbanzaicloud"}, env)))
+	logResult("add pipeline/resources/banzaicloud/configmaps.txt", addFile(tarWriter, "pipeline/resources/banzaicloud/configmaps.txt", fetchContainerCommandOutputAndError(options.cpContext, []string{"kubectl", "get", "configmaps", "-owide", "-nbanzaicloud"}, env))) // -owide does not include contents
+	logResult("add pipeline/resources/banzaicloud/secrets.txt", addFile(tarWriter, "pipeline/resources/banzaicloud/secrets.txt", fetchContainerCommandOutputAndError(options.cpContext, []string{"kubectl", "get", "secrets", "-owide", "-nbanzaicloud"}, env)))          // -owide does not include contents
+	logResult("add pipeline/resources/banzaicloud/events.yaml", addFile(tarWriter, "pipeline/resources/banzaicloud/events.yaml", fetchContainerCommandOutputAndError(options.cpContext, []string{"kubectl", "get", "events", "-oyaml", "-nbanzaicloud"}, env)))
+	// TODO add helm binary to installer image
+	logResult("add pipeline/resources/banzaicloud/helm_list.txt", addFile(tarWriter, "pipeline/resources/banzaicloud/helm_list.txt", fetchContainerCommandOutputAndError(options.cpContext, []string{"helm", "list", "--namespace", "banzaicloud", "--all"}, env)))
+
+	logResult("create pipeline/logs", addDir(tarWriter, "pipeline/logs"))
+	logResult("create pipeline/logs/banzaicloud", addDir(tarWriter, "pipeline/logs/banzaicloud"))
+	pods := strings.Split(strings.TrimSpace(fetchContainerCommandOutputAndError(options.cpContext, []string{"kubectl", "get", "pods", "-oname", "-nbanzaicloud"}, env)), "\n")
+	for _, pod := range pods {
+		pod = strings.TrimPrefix(strings.TrimSpace(pod), "pod/")
+		logResult("add pod log", addFile(tarWriter, filepath.Join("pipeline/logs/banzaicloud/", pod+".log"), fetchContainerCommandOutputAndError(options.cpContext, []string{"kubectl", "logs", "-nbanzaicloud", pod, "--all-containers"}, env)))
+	}
+
+	log.Infof("debug bundle has been written to %q", path)
+	logResult("add meta.log", addFile(tarWriter, "meta.log", logBuffer.String()))
+
+	return nil
+}
+
+// simpleCommand runs the given shell command, and returns its outputs, an error message or both
+func simpleCommand(command string, args ...string) string {
+	ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+	defer cancel()
+
+	cmd := exec.CommandContext(ctx, "sh", "-c", command)
+	if len(args) > 0 {
+		cmd = exec.CommandContext(ctx, command, args...)
+	}
```

Comment:
this way you can use *simple* shell commands instead of explicitly exec'ing commands and args (which is on the other hand safer in many cases, like in the case you commented above)